replica/database.cc

```diff
@@ -2671,7 +2660,7 @@ future<> database::flush_system_column_families() {
 future<> database::drain() {
     auto b = defer([this] { _stop_barrier.abort(); });
     // Interrupt on going compaction and shutdown to prevent further compaction
-    co_await _compaction_manager->drain();
+    co_await _compaction_manager.drain();
```

Comment:
`database::shutdown()` doesn't call `compaction_manager.stop()` anymore. So why is `drain()` still called by database?

Reply:
drain on shutdown is still a bastard.
it's called on the storage_service that in turn calls database::drain which calls compaction_manager::drain (among other things).

I think it's possible to promote the compaction_manager.drain part to be called from the upper layer, but we still want to do it out of order (i.e. not in reverse to the services start order).  Since compaction can be stopped at any time, it shouldn't be a problem to do it first before shutting down / draining other services that depend on storage.

Reply:
The drain-on-shutdown is just a boring wrapper to drain the node on shutdown if it wasn't explicitly drained before. Whether or not to put compaction_manager.drain() here or "standalone" is rather decided from -- should it be called on the `nodetool drain` or not. If not, then it's possible to take some care and move it to the upper layer. If yes -- it should stay here, at least for now.